### PR TITLE
pred + save validation images working :D

### PR DESCRIPTION
**Pull request by annikabrundyn**
by default this is taking the same 15 validation images and predicting and saving them at the end of training - I can increase the 15 to any number?

### CODE REVIEW

File: src/models/model.py [hunk 1]
```diff
@@ -88,7 +84,7 @@ def training_step(self, batch, batch_idx):
         
         # log fid
         if batch_idx % self.fid_freq == 0:
-            self._log_fid('train', pred, target)
+            self._log_fid(pred, target, step_name='train')
```
Comment (annikabrundyn) [inline src/models/model.py:87]: nothing changed - just moved around for consistency with other log function

File: src/models/model.py [hunk 2]
```diff
@@ -111,20 +107,32 @@ def validation_step(self, batch, batch_idx):
...
-            self._log_fid('valid', pred, target)
+            self._log_fid(pred, target, step_name='valid')
 
         # metrics
         ssim_val = ssim(pred, target)
         psnr_val = psnr(pred, target)
         self.log('valid_ssim', ssim_val)
         self.log('valid_psnr', psnr_val)
 
+    def test_step(self, batch, batch_idx):
```
Comment (annikabrundyn) [inline src/models/model.py:118]: at the end of training I run a "test step" but its just convenient for predicting on the val images I've kept aside to do this

File: src/models/model.py [hunk 3]
```diff
@@ -148,15 +156,22 @@ def _matplotlib_imshow_input_imgs(self, img, folder_name, frame_nums):
...
 
-    def _matplotlib_imshow_dm(self, img, title, inverse=True, cmap='magma'):
+    def _matplotlib_imshow_dm(self, img, title, inverse=True, cmap='magma', save_fig=False):
         if inverse:
             img = 1 - img
         npimg = img.squeeze().detach().cpu().numpy()
         fig = plt.figure()
         plt.imshow(npimg, cmap=cmap)
         plt.title(title)
+
```
Comment (annikabrundyn) [inline src/models/model.py:171]: for saving instead of outputting to log

File: src/models/model.py [hunk 4]
```diff
@@ -239,6 +260,9 @@ def add_model_specific_args(parent_parser):
     print("model instance created")
 
     # train
-    trainer = pl.Trainer.from_argparse_args(args)
+    trainer = pl.Trainer(callbacks=[EarlyStopping(monitor='val_loss')]).from_argparse_args(args)
```
Comment (annikabrundyn) [inline src/models/model.py]: without early stopping i have to manually kill jobs which means we'll never reach the predict/test step where we save predictions so added  lightning early stopping based on val loss. can also restrict the max epochs

File: README.md [hunk 5]
```diff
@@ -48,7 +55,7 @@ This is the UNet architecture that is used in the two different models
 e.g. using 1 gpu
 
 ```python
-python model.py --data_dir '/Users/annikabrundyn/Developer/da_vinci_depth/daVinci_data' --gpus 1 --frames_per_sample 5 --frames_to_drop 2 --batch_size 16 --lr 0.001
+python model.py --data_dir '/Users/annikabrundyn/Developer/da_vinci/daVinci_data' --gpus 1 --frames_per_sample 5 --frames_to_drop 2 --batch_size 16 --lr 0.001
```
Comment (annikabrundyn) [inline README.md:58]: i just removed the depth from the original project name to make it shorter

File: setup.py [hunk 6]
```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env python
```
Comment (annikabrundyn) [inline setup.py:1]: setup.py file is needed to import the project as a python package - just install with:

pip install -e .

(added to readme as well)

File: setup.py [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+    name='da_vinci',
+    version='0.0.0',
+    description='DaVinci Depth Estimation + Stereo View Prediction',
+    author='',
+    author_email='',
+    # REPLACE WITH YOUR OWN GITHUB PROJECT LINK
+    url='https://github.com/annikabrundyn/da_vinci',
+    install_requires=['pytorch-lightning'],
+    packages=find_packages(),
+)
```
Comment (annikabrundyn) [inline setup.py:15]: setup.py file needed to install project as python package

to install, run: pip install -e . 
(added to readme)

**Review by jeswan — APPROVED** [review]
